examples/stocks/lib/stock_home.dart

```diff
@@ -202,11 +201,7 @@ class StockHomeState extends State<StockHome> {
         Navigator.of(context).pushNamed('/stock/${stock.symbol}', mostValuableKeys: mostValuableKeys);
       },
       onShow: (Stock stock, Key arrowKey) {
-        showBottomSheet(
-          placeholderKey: _bottomSheetPlaceholderKey,
-          context: context,
-          child: new StockSymbolBottomSheet(stock: stock)
-        );
+        scaffoldKey.currentState.showBottomSheet(new StockSymbolBottomSheet(stock: stock));
```

Comment:
Ah, right. I'm not sure how much that actually matters with a bottom sheet.
But yeah, I should make it take a builder and call that from the bottom
sheet class.

On Thu, Nov 19, 2015, 20:31 Adam Barth <notifications@github.com> wrote:

> In examples/stocks/lib/stock_home.dart
> <https://github.com/flutter/flutter/pull/508#discussion_r45435017>:
>
> > @@ -202,11 +201,7 @@ class StockHomeState extends State<StockHome> {
> >          Navigator.of(context).pushNamed('/stock/${stock.symbol}', mostValuableKeys: mostValuableKeys);
> >        },
> >        onShow: (Stock stock, Key arrowKey) {
> > -        showBottomSheet(
> > -          placeholderKey: _bottomSheetPlaceholderKey,
> > -          context: context,
> > -          child: new StockSymbolBottomSheet(stock: stock)
> > -        );
> > +        scaffoldKey.currentState.showBottomSheet(new StockSymbolBottomSheet(stock: stock));
>
> If you setState the widget that contains the scaffold, the bottom sheet
> won't rebuild. It's the same issue we had with Drawer when showDrawer took
> a Widget instead of a WidgetBuilder.
>
> You're not introducing this issue in this patch, but it's probably
> something we'll want to fix in a later patch.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/flutter/flutter/pull/508/files#r45435017>.
>